find_email_address2.py

```diff
@@ -0,0 +1,62 @@
+import sys, re, urllib
+from bs4 import BeautifulSoup
+from contextlib import closing
+from selenium.webdriver import Firefox
+from selenium.webdriver.support.ui import WebDriverWait
+
+visitedDomains = set()
+
+def find_email_address2(domain):
+    #if the print stays here it prints before the check
+    if domain in visitedDomains:
+        return
+    #print("visiting: "+domain)
+    with closing(Firefox()) as browser:
+        browser.get('http://'+domain)
```

Comment:
When I ran this on jana.com and mit.edu, this at least sometimes tries to go to invalid domains like:
'http://http://web.mit.edu/initiatives/' instead of 'http://web.mit.edu/initiatives/'.  How would you fix that?